### PR TITLE
fix(windows) make ansi escape sequences render correctly

### LINKED ISSUE #711: highlighting "stuff" is showing garbage rather than what the ascii escape sequence is supposed to be doing.
The log entries at warnign level print with yellow, but other things that do not use the log but use `util.ColorInfo` produce garbage characters.

see example.
image

obviously windows releated.

```
C:\Users\jnord>jx version
NAME               VERSION
jx                 [32m1.2.20[0m
jenkins x platform [32m0.0.784[0m
kubernetes cluster [32mv1.8.8-gke.0[0m
kubectl            [32mv1.9.6[0m
helm client        [32mv2.7.2+g8478fb4[0m
helm server        [32mv2.7.2+g8478fb4[0m
git                [32mgit version 2.16.2.windows.1[0m
```

### PR DESCRIPTION
**Pull request by jtnord**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Description

On modern windows 10 (and server) variants windows supports native ansi
escape code handling.  We just need to enable it :-o

#### Special notes for the reviewer(s)

before and after

image

#### Which issue this PR fixes

fixes #711

### CODE REVIEW

**Comment by jtnord** [comment]
/hold
need to test that the survey stuff still works.

File: cmd/jx/app/jx-win.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+const (
+	// https://docs.microsoft.com/en-us/windows/console/setconsolemode
+	enableProcessedOutput = 0x1
+	enableWrapAtEOLOutput = 0x2
+	enableVirtualTermainalProcessing = 0x4
+
+)
+func configureTerminalForAnsiEscapes() {
```
Comment (agentgonzo) [inline cmd/jx/app/jx-win.go]: formatting around this method (whitespace) is a bit off

File: cmd/jx/app/jx-win.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+	kernel32  := syscall.NewLazyDLL("kernel32.dll")
```
Comment (agentgonzo) [inline cmd/jx/app/jx-win.go]: Please put a comment as to _why_ (ie, otherwise you get rubbish escape sequences printed) we are doing this

**Comment by agentgonzo** [comment]
For me, survey commands behave as they did before (ie, they work fine under windows cmd.exe, and but not under gitbash (they didn't work under gitbash before - https://github.com/jenkins-x/jx/issues/939) and you have to prepend the command with `winpty`).

**Comment by agentgonzo** [comment]
:shipit:

**Comment by jtnord** [comment]
I love a broken test in the morning.  Given we have no tests for windows in the infrastructure I call bogus result.

/test all

**Pull request merged**